// @vuepress/theme-default/lib/components/SidebarChild.js
import { h } from 'vue';
import { useRoute } from 'vue-router';
import NavLink from './NavLink.vue';
const normalizePath = (path) => decodeURI(path)
    .replace(/#.*$/, '')
    .replace(/(index)?\.(md|html)$/, '');
const isActive = (route, target) => {
    if (target === undefined) {
        return false;
    }
    if (route.hash === target) {
        return true;
    }
    const currentPath = normalizePath(route.path);
    const targetPath = normalizePath(target);
    return currentPath === targetPath;
};
const renderItem = (item, props) => {
    // if the item has link, render it as `<NavLink>`
    if (item.link) {
        return h(NavLink, {
            ...props,
            item,
        });
    }
    // if the item only has text, render it as `<p>`
    return h('p', props, item.text);
};
const renderChildren = (item, depth) => {
    var _a;
    if (!((_a = item.children) === null || _a === void 0 ? void 0 : _a.length)) {
        return null;
    }
    return h('ul', {
        class: {
            'sidebar-sub-headers': depth > 0,
        },
    }, item.children.map((child) => h('li', h(SidebarChild, {
        item: child,
        depth: depth + 1,
    }))));
};
export const SidebarChild = ({ item, depth }) => {
    var _a;
    if (item.isGroup) {
        return [
            h('section', {
                class: {
                    'sidebar-group': true,
                },
            }, [
                renderItem(item, {
                    class: 'sidebar-heading',
                }),
                renderChildren(item, depth),
            ]),
        ];
    }
    const route = useRoute();
    const active = isActive(route, item.link) || ((_a = item.children) === null || _a === void 0 ? void 0 : _a.some((child) => isActive(route, child.link)));
    return [
        renderItem(item, {
            class: {
                'sidebar-link': true,
                active,
            },
        }),
        renderChildren(item, depth),
    ];
};
SidebarChild.displayName = 'SidebarChild';
SidebarChild.props = {
    item: {
        type: Object,
        required: true,
    },
    depth: {
        type: Number,
        required: false,
        default: 0,
    },
};
